drip/tokens.py

```diff
@@ -0,0 +1,119 @@
+from typing import Optional, Tuple
+
+from django.contrib.auth.models import AbstractBaseUser
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.utils.crypto import constant_time_compare
+from django.utils.encoding import force_bytes, force_str
+from django.utils.http import base36_to_int, urlsafe_base64_decode, urlsafe_base64_encode
+
+from drip.models import Drip
+from drip.utils import get_user_model
+
+User = get_user_model()
+
+
+class CustomTokenGenerator(PasswordResetTokenGenerator):
+    """
+    Custom token generator with no login and password hashing.
+    Also not checking time validation, it is just a hash of user data for email purposes.
+    """
+
+    def _make_hash_value(self, user: AbstractBaseUser, timestamp: int) -> str:
+        """
+        Build hash value ignoring login_timestamp and password for keeping it valid over time
+        """
+        email_field = user.get_email_field_name()
+        email = getattr(user, email_field, "") or ""
```

Comment:
I'm asking because I'm not sure: do you need the `or ""`?

Reply:
I copy this code directly from Django source code but I remove the time check. I will leave it as it is right now.